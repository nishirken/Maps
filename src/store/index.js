export testInitialState from './test-initial-state';
export testServerInitialState from './test-server-initial-state';
export store from './create-store';
export testStore from './test-store';
