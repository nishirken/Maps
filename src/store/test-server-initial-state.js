import { fromJS } from 'immutable';

export default fromJS([
    {
        index: 1,
        coords: {
            lng: 100,
            lat: 200,
            x: 20,
            y: 21,
        },
        name: 'Marker 1',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
    {
        index: 2,
        coords: {
            lng: 200,
            lat: 300,
            x: 10,
            y: 11,
        },
        name: 'Marker 2',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
]);
