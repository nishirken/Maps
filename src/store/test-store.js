import configureStore from 'redux-mock-store';
import initialState from './test-initial-state';

const mockStore = configureStore();
const store = mockStore(initialState.toObject());

export default store;
