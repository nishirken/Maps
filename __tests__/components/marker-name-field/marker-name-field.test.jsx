import store from 'Store';
import { MarkerNameField } from 'Components';

describe('Marker name field', () => {
    const TestMarkerNameFieldShallow = shallow(
        <MarkerNameField
            store={store}
            x={100}
            y={99}
        />
        ),
        TestMarkerNameField = mount(
            <MarkerNameField
                store={store}
                x={100}
                y={99}
            />
        );

    it('Rendered', () => {
        expect(TestMarkerNameFieldShallow).toMatchSnapshot();
    });

    it('Has a title', () => {
        expect(TestMarkerNameField.find('h3').length).toBe(1);
    });

    it('Has an input', () => {
        expect(TestMarkerNameField.find('input').length).toBe(1);
    });
});
