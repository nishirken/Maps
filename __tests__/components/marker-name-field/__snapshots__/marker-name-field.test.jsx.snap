// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker name field Not rendered, when has a completeCreateMarkerWithoutName state 1`] = `null`;

exports[`Marker name field Rendered 1`] = `
<styled.div
  x={100}
  y={99}
>
  <styled.h3>
    Please enter a name for your marker
  </styled.h3>
  <styled.input
    innerRef={[Function]}
    onKeyDown={[Function]}
    type="text"
  />
</styled.div>
`;
